Wikipedia/Code/WMFDatabaseHouseKeeper.swift

```diff
@@ -6,9 +6,31 @@ import Foundation
     @objc func performHouseKeepingOnManagedObjectContext(_ moc: NSManagedObjectContext) throws -> [URL] {
         
         let urls = try deleteStaleUnreferencedArticles(moc)
+        try deleteStaleTalkPages(moc)
         return urls
     }
 
+    /**
+     
+     We only persist the last 50 most recently accessed talk pages, delete all others.
+     
+    */
+    private func deleteStaleTalkPages(_ moc: NSManagedObjectContext) throws {
+        let request: NSFetchRequest<TalkPage> = TalkPage.fetchRequest()
+        request.sortDescriptors = [NSSortDescriptor(key: "dateAccessed", ascending: false)]
+        request.fetchOffset = 50
+        
+        let oldTalkPages = try moc.fetch(request)
+        for talkPage in oldTalkPages {
+            moc.delete(talkPage)
```

Comment:
might be a good use case for a batch delete https://developer.apple.com/library/archive/featuredarticles/CoreData_Batch_Guide/BatchDeletes/BatchDeletes.html